### ISSUE TITLE
broken propagation of traces from OpenTelemetry tracer to Kafka producer in 1.21.0 version

### ISSUE THREAD

**Issue by RuslanHryn** [DataDog/dd-trace-java#6042]
Propagation traces from OpenTelemetry tracer to Kafka producer stop working in dd-trace-java 1.21.0 version
This functionality has worked fine in previous versions of d-trace-java

**To reproduce**:
Enable OpenTelemetry instrumentation by property `dd.trace.otel.enabled=true`
Use dd-trace-java 1.21.0 version
Start a new span using OpenTelemetry Tracer and send a Kafka message in this span

```
   Span parentSpan = tracer.spanBuilder("parent").startSpan();
    try (Scope scope = parentSpan.makeCurrent()) {
         kafkaProducer.send(record);
    } finally {
      parentSpan.end();
    }
```
Open APM -> Traces and parent span won't be connected with the span of the Kafka producer.

**Comment by mcculls** [comment]
Hi @RuslanHryn - this should be fixed in the upcoming 1.22.0 release

You can try the latest snapshot from https://oss.sonatype.org/content/repositories/snapshots/com/datadoghq/dd-java-agent/1.22.0-SNAPSHOT/dd-java-agent-1.22.0-20231012.161439-114.jar to confirm

**Comment by RuslanHryn** [comment]
@mcculls 
Thanks, I will test it on snapshot version and let you

**Comment by RuslanHryn** [comment]
@mcculls 
I've tested it and can confirm that the issues was fixed in 1.22.0
Thanks, I will close this issue
